MoreLinq/InsertAt.cs

```diff
@@ -0,0 +1,70 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2017 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+
+    static partial class MoreEnumerable
+    {
+
+        /// <summary>
+        /// Inserts the elements of a sequence into another sequence, at the specified index.
+        /// </summary>
+        /// <typeparam name="T">Type of the elements of the source sequence.</typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="second">The sequence that will be inserted into source.</param>
+        /// <param name="index">The zero-based index at which the new elements should be inserted.</param>
+        /// <returns>
+        /// A sequence that contains the elements of <paramref name="source"/>
+        /// Plus the elements of <paramref name="second"/>, at the specified index.
+        /// </returns>
+        /// <exception cref="ArgumentOutOfRangeException">
```

Comment:
There's the `ArgumentNullException` notes missing. Also, split this into two `<exception>` tags, one for each condition.